temporal-sdk/src/main/java/io/temporal/client/BuildIDOperation.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.client;
+
+import io.temporal.api.workflowservice.v1.UpdateWorkerBuildIdCompatibilityRequest;
+import javax.annotation.Nonnull;
+
+/**
+ * The implementations of this class can be passed as parameters to {@link
+ * WorkflowClient#updateWorkerBuildIDCompatability(String, BuildIDOperation)}
+ *
+ * <p>See each public static method to learn about and construct the available operations.
+ */
+public abstract class BuildIDOperation {
```

Comment:
Yeah I dislike this about the Java SDK, but it is pretty consistent  with not capitalizing both letter so we should  be too.

Reply:
Guess I'll rename it all. I actually prefer `Id` to `ID`. `ID` is not an acronym, the `D` doesn't stand for anything by itself. Not sure on what axis `ID` makes more sense, but I guess there are some strong feelings there lol. I was just sticking with what I'd done in Go, but should've just looked around.

Reply:
I personally like this about the Java SDK and in general. Otherwise you can't tell boundaries of consecutive acronyms/abbreviations like `XMLRPC` or `GRPC`. In .NET it's "all upper if only two letters, only first if more" which is a little weird too.